lib/msf/core/encoded_payload.rb

```diff
@@ -237,7 +237,7 @@ def encode
 
           begin
             eout = self.encoder.encode(eout, reqs['BadChars'], nil, pinst.platform)
-          rescue EncodingError
+          rescue EncodingError => e
             wlog("#{err_start}: Encoder #{encoder.refname} failed: #{$!}", 'core', LEV_1)
```

Comment:
Maybe explicitly print this error now that we explicitly catch it?  Also switch out the environment variable below with the explicit back trace call.